distros/generate.sh

```diff
@@ -9,7 +9,13 @@ set -e
 #	or: ./generate.sh debian-newversion
 #		to create a new folder and a Dockerfile within it
 
-cd "$(dirname "$(readlink -f "$BASH_SOURCE")")"
+# `readlink -f` does not work on macOS.
+# To get `greadlink`, run `brew install coreutils`.
+if [ "$(uname)" == "Darwin" ]; then
```

Comment:
https://github.com/dcos/dcos/blob/b5f75d38e85e26ead42a1c41ec70c7c09591c6c0/gen/build_deploy/bash/dcos_generate_config.sh.in#L19 is the line that caused an issue. With this PR and the instructions in it, we can run DC/OS Docker on macOS if that answers your question.